docs/src/CommutativeAlgebra/localizations.md

```diff
@@ -111,11 +111,14 @@ This reflects the way of creating localizations of quotients of multivariate pol
 julia> R, (x, y, z) = polynomial_ring(QQ, ["x", "y", "z"]);
 
 julia> P = ideal(R, [x])
-ideal(x)
+Ideal
+  of multivariate polynomial ring in 3 variables over QQ
+generated by
+  x
 
 julia> U = complement_of_prime_ideal(P)
 Complement
-  of prime ideal(x)
+  of prime ideal with 1 generator in multivariate polynomial ring
   in multivariate polynomial ring in 3 variables over QQ
```

Comment:
In places like this, we loose information while now printing redundant information. Don't get me wrong, at the same time there are good reasons for the change in this PR (overall I like it!).

But I wonder if that means we perhaps need another printing mode, or just some special cases where we don't just recurse but do something specific when recording? 

Like here, perhaps the complement function should print a list of generators of the ideal? If the printing here was any of these, I think it would be better (but perhaps not in general?)
```
Complement of
  prime ideal
    of multivariate polynomial ring in 3 variables over QQ
  generated by
    x
```
or
```
Complement of
  prime ideal generated by
    x
in multivariate polynomial ring in 3 variables over QQ
```
Of course in this specific case some people might also like
```
Complement of
  prime ideal P
in R
```
or
```
Complement of
  prime ideal generated by [ x ]
  in QQ[x, y, z]
```

but both also have downsides, IMHO...


Perhaps @simonbrandhorst or @fieker or @thofma or others have something to say...